process/block/shardblock.go

```diff
@@ -1710,20 +1734,28 @@ func (sp *shardProcessor) receivedMetaBlock(headerHandler data.HeaderHandler, me
 
 		// attesting something
 		if sp.hdrsForCurrBlock.missingHdrs == 0 {
-			sp.hdrsForCurrBlock.missingFinalityAttestingHdrs = sp.requestMissingFinalityAttestingHeaders(
-				core.MetachainShardId,
-				sp.metaBlockFinality,
-			)
+			if !sp.shouldConsiderProofsForNotarization(metaBlock) {
```

Comment:
on the else, you should at least check that there is a valid proof for this metablock, otherwise add it to the missing proofs.

Reply:
added